cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,659 @@
+        fig = plt.figure()
+        plt.plot(self._num_cfds_seq, self._gnd_state_probs, 'ro-',
+                 figure=fig, **plot_kwargs)
+        plt.xlabel(r"Number of Cliffords", figure=fig)
+        plt.ylabel('Ground State Probability', figure=fig)
+        fig.show()
+
+
+class TomographyResult:
+    """Results from a state tomography experiment."""
+
+    def __init__(self, density_matrix: np.ndarray):
+        """
+        Args:
+            density_matrix: The density matrix obtained from tomography.
+        """
+        self._density_matrix = density_matrix
+
+    @property
+    def data(self) -> np.ndarray:
+        """Returns an n^2 by n^2 complex matrix representing the density
+        matrix of the n-qubit system.
+        """
+        return self._density_matrix
+
+    def plot(self) -> None:
+        """Plots the real and imaginary parts of the density matrix as two
+        3D bar plots.
+        """
+        fig_re, fig_imag = _plot_density_matrix(self._density_matrix)
+        fig_re.show()
+        fig_imag.show()
+
+
+def rabi_oscillations(sampler: sim.SimulatesSamples, qubit: devices.GridQubit,
+                      max_angle: float = 2 * np.pi, *,
+                      repetitions: int = 1000,
+                      num_points: int = 200) -> RabiResult:
+    """Runs a Rabi oscillation experiment.
+
+    Rotates a qubit around the x-axis of the Bloch sphere by a sequence of Rabi
+    angles evenly spaced between 0 and max_angle. For each rotation, repeat
+    the circuit a number of times and measure the average probability of the
+    qubit being in the |1> state.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        max_angle: The final Rabi angle in radians.
+        repetitions: The number of repetitions of the circuit for each Rabi
+            angle.
+        num_points: The number of Rabi angles.
+
+    Returns:
+        A RabiResult object that stores and plots the result.
+    """
+    theta = sympy.Symbol('theta')
+    circuit = circuits.Circuit.from_ops(ops.X(qubit) ** theta)
+    circuit.append(ops.measure(qubit, key='z'))
+    sweep = study.Linspace(key='theta', start=0.0, stop=max_angle / np.pi,
+                           length=num_points)
+    results = sampler.run_sweep(circuit, params=sweep, repetitions=repetitions)
+    angles = np.linspace(0.0, max_angle, num_points)
+    excited_state_probs = np.zeros(num_points)
+    for i in range(num_points):
+        excited_state_probs[i] = np.mean(results[i].measurements['z'])
+
+    return RabiResult(angles, excited_state_probs)
+
+
+def single_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                         qubit: devices.GridQubit,
+                                         num_cliffords: Sequence[int],
+                                         use_xy_basis: bool = True,
+                                         *, num_circuits: int = 20,
```

Comment:
Put the * before `num_cliffords`, and on its own line.

Reply:
Done. Same modification applied to the arguments of `two_qubit_randomized_benchmarking`